keras_cv/layers/preprocessing/grid_mask.py

```diff
@@ -0,0 +1,148 @@
+        image_height = tf.cast(image_height, tf.float32)
+        image_width = tf.cast(image_width, tf.float32)
+
+        mask_w = mask_h = tf.cast(
+            tf.cast((self.gridmask_size_ratio + 1), tf.float32)
+            * tf.math.maximum(image_height, image_width),
+            tf.int32,
+        )
+
+        mask = tf.zeros(shape=[mask_h, mask_w], dtype=tf.int32)
+        gridblock = tf.random.uniform(
+            shape=[],
+            minval=int(tf.math.minimum(image_height * 0.5, image_width * 0.3)),
+            maxval=int(tf.math.maximum(image_height * 0.5, image_width * 0.3)) + 1,
+            dtype=tf.int32
+        )
+
+        if self.ratio == 1:
+            length = tf.random.uniform(
+                shape=[], minval=1, maxval=gridblock + 1, dtype=tf.int32
+            )
+        else:
+            length = tf.cast(
+                tf.math.minimum(
+                    tf.math.maximum(
+                        int(tf.cast(gridblock, tf.float32) * self.ratio + 0.5), 1
+                    ),
+                    gridblock - 1,
+                ),
+                tf.int32,
+            )
+
+        for _ in range(2):
+            start_w = tf.random.uniform(
+                shape=[], minval=0, maxval=gridblock + 1, dtype=tf.int32
+            )
+
+            for i in range(mask_w // gridblock):
+                start = gridblock * i + start_w
+                end = tf.math.minimum(start + length, mask_w)
+                indices = tf.reshape(tf.range(start, end), [end - start, 1])
+                updates = (
+                    tf.ones(shape=[end - start, mask_w], dtype=tf.int32)
+                    * self.fill_value
+                )
+                mask = tf.tensor_scatter_nd_update(mask, indices, updates)
+            mask = tf.transpose(mask)
+        return mask
+
+    @tf.function
+    def _grid_mask(self, image, training):
+        image_height = tf.shape(image)[0]
+        image_width = tf.shape(image)[1]
+        grid = self.mask(image_height, image_width)
+        # TODO: rnadomly rotate the grid, i.e. tf.image.rot90
+
+        mask = tf.reshape(
+            tf.cast(self.crop(grid, image_height, image_width), image.dtype),
+            (image_height, image_width),
+        )
+        mask = tf.expand_dims(mask, -1) if image._rank() != mask._rank() else mask
+
+        rate_cond = tf.less(
+            tf.random.uniform(shape=[], minval=0, maxval=1.0), self.rate
+        )
+        augment_cond = tf.logical_and(rate_cond, training)
+        return tf.cond(augment_cond, lambda: image * mask, lambda: image)
+
+    def call(self, images, training=True):
+        """Masks input image tensor with random grid mask."""
+        if training is None:
+            training = backend.learning_phase()
+        
+        # TODO: Make the batch operation vectorize.
+        return tf.map_fn(lambda image: self._grid_mask(image, training), images)
```

Comment:
if not training:
    return images

then you can drop the training param

Reply:
How about like this?

```python
 def call(self, images, training=None):
     """Masks input image tensor with random grid mask."""
     if training:
        return tf.map_fn(lambda image: self._grid_mask(image, training), images)
     return images
```

Reply:
Typically guard clauses scale better.  The default case should be the final return.

so let's just switch the if training and the image return statements.

Additionally, include the None check in training (see RandomCutout).  

Thanks!